app/src/main/java/com/example/development/sakaiclientandroid/utils/custom/CustomLinkMovementMethod.java

```diff
@@ -0,0 +1,149 @@
+        fragmentManager = new WeakReference<>(manager);
+    }
+
+    /**
+     * Largely pulled from the Android source code except for the logic
+     * of opening a {@code WebView} when a link is clicked.
+     * @param widget TextView in which the clicked span resides
+     * @param buffer The touched buffer
+     * @param event The type of touch event (eg. up, down, left, right, etc.)
+     * @return True if the touch event was handled manually, otherwise whatever
+     *          super(widget, buffer, event) returns.
+     */
+    @Override
+    public boolean onTouchEvent(TextView widget, Spannable buffer, MotionEvent event) {
+        int action = event.getAction();
+
+        // Only open webview if something was clicked, instead of swiping left/right
+        if (action == MotionEvent.ACTION_UP || action == MotionEvent.ACTION_DOWN) {
+            // Get the location of the click
+            int x = (int) event.getX();
+            int y = (int) event.getY();
+
+            // Adjust location for padding and scroll distance
+            x -= widget.getTotalPaddingLeft();
+            y -= widget.getTotalPaddingTop();
+
+            x += widget.getScrollX();
+            y += widget.getScrollY();
+
+            // Get the line at the calculated area
+            Layout layout = widget.getLayout();
+            int line = layout.getLineForVertical(y);
+            int off = layout.getOffsetForHorizontal(line, x);
+
+            // Get the link(s) clicked based on the selection location
+            ClickableSpan[] links = buffer.getSpans(off, off, ClickableSpan.class);
+
+            if (links.length != 0) {
+                if (action == MotionEvent.ACTION_UP) {
+                    openWebView(links[0]);
+                } else { // action == MotionEvent.ACTION_DOWN
+                    Selection.setSelection(buffer,
+                            buffer.getSpanStart(links[0]),
+                            buffer.getSpanEnd(links[0]));
+                }
+                return true;
+            } else {
+                Selection.removeSelection(buffer);
+            }
+        }
+
+        return super.onTouchEvent(widget, buffer, event);
+    }
+
+
+    /**
+     * Using the {@code fragmentManager}, creates a {@code WebFragment}
+     * that opens up the given link
+     * @param clickedLink The link that was clicked and should be opened
+     */
+    private void openWebView(ClickableSpan clickedLink) {
+        // Check that the Clickable link is a URL Link
+        // (A URLSpan is automatically created with 'a' tags from
+        // HTML)
+        if( !(clickedLink instanceof URLSpan) )
+            return;
+
+        URLSpan clicked = (URLSpan) clickedLink;
+        String link = clicked.getURL();
+
+        // If a fragment manager is available, create the WebView to view the
+        // link
+        if(fragmentManager != null && fragmentManager.get() != null) {
```

Comment:
Purposefully didn't include an `else` statement because if it is null then the `FragmentManager` has been GC'ed, implying that the app is no longer in the foreground anyways.